iree/tools/iree-benchmark-module-main.cc

```diff
@@ -35,7 +35,8 @@ ABSL_FLAG(std::string, module_file, "",
 
 ABSL_FLAG(std::string, entry_function, "",
           "Name of a function contained in the module specified by module_file "
-          "to run.");
+          "to run. If this is not set, all the exported functions will be  "
+          "benchmarked and they are expected to not have input arguments");
```

Comment:
This seems like a good incremental improvement, but longer term it would be nice to be able to specify inputs if desired regardless of whether we're benchmarking all exported functions or only some of them. We should also probably always register a benchmark for each exported function and use the benchmark library's built-in benchmark filtering flag to determine which to run.

Reply:
Yes, I think it's better to specify inputs if we're benchmarking all exported functions. The function_input is only needed when user specify entry_function, IMO. Let's file an issue to track it, what do you think?

Reply:
The assumption is the compiler is going to generate a `benchmark_foo(void)`. The majority of the cases the benchmarks aren't sensitive to inputs like all dense math! only sparse or indexing op e.g gather. so a hook that just run everything with no entries and when a specific benchmark needs a hand inputs we can use entry_function with input.

Reply:
Yup I think allowing no-input is good (Oleg asked for something similar for iree-run-mlir), but the API here is a bit wonky with this combination. We should be able to make it a lot simpler while leaving the "I don't care about the inputs" case easy. Filing an issue SGTM